idl/response/foxsi_get_2d_image.pro

```diff
@@ -0,0 +1,148 @@
+;;; 
+;;;
+;;;CALL SEQUENCE: rebinned_convolved_map = foxsi_get_2d_image()
+;;;
+;;;
+;;;KEYWORDS:      source_map = "source_map". User inputted source, if
+;;;               blank, default generated from get_source_map
+;;;               function
+;;;
+;;;               px = "pixel size of detector" in arcseconds, default is 3''
+;;;
+;;;
+;;;COMMENTS:      -Runtime scales badly with FOV size
+;;;               -The default source array is 
+;;;               set to [150,150] ~2.5' X 2.5' FOV at 1 arcsec per
+;;;               pixel, this takes a few seconds to run.
+;;;               -Convolution method improved - slight speed
+;;;                upgrade,shorter and more accurate. Removes hard
+;;;                edges and asymmetry which were previously
+;;;                detectable at low resolutions.
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+FUNCTION foxsi_get_2d_image,source_map = source_map, px = pix_size
+
+IF N_ELEMENTS(SOURCE_MAP) EQ 0 THEN PRINT, 'No user input detected, using default source image'
+
+;;;;; Check for updates to peripheral functions for the purposes of testing
+RESOLVE_ROUTINE, 'foxsi_get_psf_array', /IS_FUNCTION
+RESOLVE_ROUTINE, 'foxsi_get_source_map', /IS_FUNCTION
+
+;;;; Define default source_map input in case of no user input
+DEFAULT, source_map, foxsi_get_source_map()
+
+;;;;; Define default detector resolution to 3 arcsecs per pixel
+DEFAULT, pix_size, 3
+
+x=0 ;; Redundant FOV coordinates required as arguments for get_psf_array
+y=0 ;; Will come into play when FOV coordinate dependence is introduced into psf
+
+x_size = N_ELEMENTS(REFORM(source_map.data[*,0]))*1.0  ;;; Get dimensions of FOV in pixels
+y_size = N_ELEMENTS(REFORM(source_map.data[0,*]))*1.0 
+
+print, strcompress('Source_Array_is_'+string(N_ELEMENTS(REFORM(source_map.data[*,0]))) $
+       +'x'+string(N_ELEMENTS(REFORM(source_map.data[0,*])))+'_Pixels', /REMOVE_AL)
+
+;;; Interpolate source_map to achieve odd dimensions for the purpose
+;;; of accurate convolution
+
+o_x_size =  x_size + 1 - (x_size MOD 2)
+o_y_size =  y_size + 1 - (y_size MOD 2)
+
+odd_source_array = CONGRID(source_map.data,o_x_size,o_y_size) 
+
+;;; Obtain psf assuming constant across FOV
+
+psf_array = foxsi_get_psf_array(source_map.xc,source_map.yc,source_map.dx      $
+,source_map.dy,x,y,x_size=o_x_size,y_size=o_y_size) 
+
+psf_x_size = n_elements(reform(psf_array[*,0]))*1.0
+psf_y_size = n_elements(reform(psf_array[0,*]))*1.0
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+
+;;; 08/31 -Improved convolution method takes only the pixels in the psf which
+;;; overlap with the convolved image for a given FOV pixel. 
+;;; Eliminates need to shift whole array and therefore the problem
+;;; with wrapping.
+
+
+convolved_array = DBLARR(o_x_size,o_y_size)
+
+FOR i = 0.0, N_ELEMENTS(odd_source_array)-1 DO BEGIN
+
+        x = (i MOD o_x_size)*1.0  
```

Comment:
We can't assume that all users will have moved to 8.5.

Why not just use a nested `for` loop since you already know both ranges?  (It's IDL, so it'll be just as fast, if not slightly faster, and the 1D index `i` isn't being used for any fancy access.)